modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -271,6 +277,97 @@
         (hx/->AtTimeZone (zone-id->windows-zone target-timezone))
         hx/->datetime)))
 
+(defmethod sql.qp/->honeysql [:sqlserver :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)
+        disallowed-types (keep
```

Comment:
I wonder whether if `x` or `y` doesn't have type info, does this fail?

Reply:
No, because we're using `some->`. `hx/type-info` and `hx/is-of-type?` would return `nil`.